### PR TITLE
feat: improve retrieval system

### LINKED ISSUE #4: feat(tests): leverage `wandb` for testing
Atm we run tests without any `wandb` tracing, however it'd be much better if all CI runs have `wandb` tracing allowing for better visibility.

## Tasks
- [x] Setup a `wandb` key as a secret for actions.
- [x] Enable capturing stdout in tests so that wandb traces can be inspected from actions logs. ([b102ed5](https://github.com/soumik12345/ml-framework-bot/pull/16/commits/b102ed571cf091df010a37b7ffcf56fbb6dfce31))

### LINKED ISSUE #7: feat(eval): Create an evaluation strategy for evaluating `KerasDocumentationAgent`
- [x] Formulate an evaluation strategy to evaluate `KerasDocumentationAgent`.
- [x] Create an evaluation dataset according to the formulated evaluation strategy. (Available as `keras_evaluation_dataset:v0`)
- [x] Implement the evaluation strategy using [`weave.Evaluate`](https://weave-docs.wandb.ai/guides/core-types/evaluations).

### PR DESCRIPTION
**Pull request by SauravMaheshkar**
* Adds a mapping from API to references
* switches retrieval system to one based on mappings

### CODE REVIEW

File: ml_frameworks_bot/keras/docs_agent.py [hunk 1]
```diff
@@ -19,21 +24,21 @@ class KerasOpWithAPIReference(BaseModel):
 class KerasDocumentationAgent(weave.Model):
     op_extraction_llm_client: LLMClientWrapper
     retrieval_augmentation_llm_client: LLMClientWrapper
```
Comment (soumik12345) [inline ml_frameworks_bot/keras/docs_agent.py:26]: We probably can skip the `ask_llm_about_op` operation using the `retrieval_augmentation_llm_client` when using the `KerasDocumentationHeuristicRetreiver`, right?

**Review by soumik12345 — CHANGES_REQUESTED** [review]
Great work @SauravMaheshkar!
I just have 1 minor feedback.

File: tests/unit/keras/test_neural_retriever.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+from dotenv import load_dotenv
+
+from ml_frameworks_bot.keras import KerasDocumentationRetreiver
+
+load_dotenv()
+
+
+def test_keras_retriever():
```
Comment (soumik12345) [inline tests/unit/keras/test_neural_retriever.py:8]: We should also disable this from running in the CI.

**Review by soumik12345 — CHANGES_REQUESTED** [review]

File: ml_frameworks_bot/keras/docs_agent.py (hunk 1 above)
Comment (SauravMaheshkar) [reply ml_frameworks_bot/keras/docs_agent.py:26]:
```python
is_neural_retriever = isinstance(
    self.api_reference_retriever, 
    KerasDocumentationRetreiver
)
if is_neural_retriever:
    purpose_of_op = self.ask_llm_about_op(keras_op)
    api_reference: BaseNode = self.api_reference_retriever.predict(
        query=f"API reference for `{keras_op}`.\n{purpose_of_op}",
    )[0]
else:
    api_reference: BaseNode = self.api_reference_retriever.predict(
        query=keras_op
    )
```

Yes, we skip the `ask_llm_about_op` call when using the heuristic retriever.

File: ml_frameworks_bot/keras/eval.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
+import weave
+from weave import Dataset
+
+# Initialize Weave
+weave.init(project_name="ml-colabs/ml-frameworks-bot")
+
+# Create a dataset
+eval_dataset = Dataset(
```
Comment (soumik12345) [inline ml_frameworks_bot/keras/eval.py]: You should simply publish the dataset and use its `ref` for evaluation.
Like this
```python
dataset = weave.Dataset(name="awesome-ds", rows=[...])
weave.publish(dataset)
```

**Review by soumik12345 — CHANGES_REQUESTED** [review]

**Pull request merged**